PARCEL_2_RFC.md

```diff
@@ -0,0 +1,1285 @@
+your asset is targetting.
+
+You can configure environments through your targets.
+
+```json
+{
+  "targets": {
+    "main": {
+      "node": ">=4.x",
+      "electron": ">=2.x",
+      "browsers": ["> 1%", "not dead"]
+    }
+  }
+}
+```
+
+
+When one asset depends on another, the environment is inherited from its
+parent. But how you depend on the asset can change some properties of that
+environment.
+
+For example:
+
+```js
+navigator.serviceWorker.register('./service-worker.js');
+```
+
+```js
+let childEnvironment = { ...parentEnvironment, browserContext: 'service-worker' }
+```
+
+### Caching
+
+Parcel will create a  `/node_modules/.cache/parcel` directory
+
+The top-level directory will be filled with directories with two letters, which
+are the start of a hash which is finished by the names of the JSON files inside.
+
+```
+/node_modules/.cache/parcel/
+  /00/
+    213debd8ddd45819b79a3a974ed487.json
+    40ae9b581afc53841307a4b3c2463d.json
+    63a9dd58fc1e8f8bb819759ea9793c.json
+    ...
+  /01/
+  /../
+  /zy/
+  /zz/
+```
+
+It follows this weird structure in order to avoid too many files being created
+in a single directory which degrades file system performance.
+
+## Asset Resolution
+
+Parcel follows the Node module resolution algorithm with a few additions.
+
+### Local Paths
+
+```
+./path/to/file
+./path/to/file.js
+```
+
+These follow the [Node module resolution algorithm](https://nodejs.org/api/modules.html#modules_all_together).
+
+### Package Paths
+
+```
+preact
+lodash/cloneDeep
+@sindresorhus/is
+```
+
+These follow the [Node module resolution algorithm](https://nodejs.org/api/modules.html#modules_all_together).
+
+### URLs
+
+```
+https://unpkg.com/preact@8.2.9/dist/preact.min.js
+```
+
+Parcel by default will ignore URL dependencies, other resolver plugins may
+choose to do something with them.
+
+### Tilde Paths
+
+```
+~/src/file.js
+```
+
+Only when used outside of `node_modules` directories, the `~` is replaced by an
+absolute path to the closest package root:
+
+```sh
+/path/to/app #(/package.json)
+```
+
+To form a path that looks like:
+
+```
+/path/to/app/src/file.js
+```
+
+Then it follows the [Node module resolution algorithm](https://nodejs.org/api/modules.html#modules_all_together).
+
+### Aliases
+
+Aliases come in two forms:
+
+1. Package Aliases: `react -> preact`
+2. File/Directory Aliases: `utils` -> `./src/utils`
+
+```json
+{
+  "name": "my-project",
+  "alias": {
+    "react": "preact-compat",
+    "react-dom": "preact-compat",
+    "utils": "./src/utils",
+    "components": "./src/components"
+  }
+}
+```
+
+There are a couple of rules:
+
+1. Aliases will only be respected when specified outside of `node_modules`.
+2. Aliases specified outside of `node_modules` will affect assets inside of `node_modules`.
+3. Aliases cannot build off of other aliases.
+4. Only one alias will be applied at a time.
+5. Aliases must be valid npm package names.
+
+## Plugins
+
+### Resolvers
```

Comment:
Just want to confirm a resolver plugin can do what TypeScript [module resolution spec](https://www.typescriptlang.org/docs/handbook/module-resolution.html) describes.